sklearn/utils/_array_api.py

```diff
@@ -232,6 +232,12 @@ def take(self, X, indices, *, axis=0):
     def isdtype(self, dtype, kind):
         return isdtype(dtype, kind, xp=self._namespace)
 
+    def searchsorted(self, a, v, *, side="left", sorter=None):
+        a = _convert_to_numpy(a, xp=self._namespace)
```

Comment:
The same logic can be applied for `_isin`, because pytorch has function `isin`, which works for their tensors.

Reply:
This might be dangerous in general though because different libraries might have slightly different calling conventions for methods part of their namespace but not part of the Array API spec.

So maybe we could hold a per-namespace whitelist of known-to-work Array API extensions. But I am not sure we want to start maintaining such a whitelist, in case we observe behavioral or API changes across different versions of those libraries.

Reply:
Maybe we can limit ourselves to a few cases that really matter from a performance point of view and use the safe route (numpy conversion) otherwise.